graspy/inference/latent_distribution_test.py

```diff
@@ -141,27 +198,23 @@ def fit(self, A1, A2):
 
         Returns
         -------
-        p_ : float
+        p_value : float
             The p value corresponding to the specified hypothesis test
         """
         A1 = import_graph(A1)
         A2 = import_graph(A2)
-        # if not is_symmetric(A1) or not is_symmetric(A2):
-        #     raise NotImplementedError()  # TODO asymmetric case
-        if self.n_components is None:
-            # get the last elbow from ZG for each and take the maximum
-            num_dims1 = select_dimension(A1)[0][-1]
-            num_dims2 = select_dimension(A2)[0][-1]
-            self.n_components = max(num_dims1, num_dims2)
 
         X1_hat, X2_hat = self._embed(A1, A2)
-        X1_hat, X2_hat = self._median_heuristic(X1_hat, X2_hat)
-        U = self._statistic(X1_hat, X2_hat)
-        null_distribution = self._bootstrap(X1_hat, X2_hat, self.n_bootstraps)
-        self.null_distribution_ = null_distribution
-        self.sample_T_statistic_ = U
-        p_value = (len(null_distribution[null_distribution >= U])) / self.n_bootstraps
-        if p_value == 0:
-            p_value = 1 / self.n_bootstraps
-        self.p_ = p_value
-        return self.p_
+        X1_hat, X2_hat = self._median_sign_flips(X1_hat, X2_hat)
+
+        x = np.array(X1_hat)
+        y = np.array(X2_hat)
+
+        data = self.test.test(
+            x, y, reps=self.n_bootstraps, workers=self.num_workers, auto=False
+        )
+        self.sample_T_statistic_ = data[0]
+        self.p_value_ = data[1]
+        self.null_distribution_ = self.test.indep_test.null_dist
+
+        return self.p_value_
```

Comment:
maybe can address later but im realizing this should really return an instance of `self`, thoughts? just for sklearn consistency and consistency within graspy even

Reply:
@bdpedigo I have been thinking about this for a while while working on my fix, and I entirely agree with you. this should return self and we can have a fit_predict.

Reply:
awesome. let's leave it for now and I will make an issue because we will have to change semipar also

Reply:
https://github.com/neurodata/graspy/issues/354
issue. resolving